dotnet/src/Functions/Functions.OpenAPI/OpenApi/OpenApiDocumentParser.cs

```diff
@@ -287,6 +287,21 @@ private static List<RestApiOperationParameter> CreateRestApiOperationParameters(
         return new RestApiOperationPayload(mediaType, payloadProperties, requestBody.Description, mediaTypeMetadata?.Schema?.ToJsonDocument());
     }
 
+    private static IEnumerable<(string, RestApiOperationResponse)> CreateRestApiOperationResponses(OpenApiResponses responses)
+    {
+        foreach (var response in responses)
+        {
+            var mediaType = s_supportedMediaTypes.FirstOrDefault(smt => response.Value.Content.ContainsKey(smt));
+            if (mediaType is not null)
+            {
+                var matchingSchema = response.Value.Content[mediaType].Schema;
+                var description = response.Value.Description ?? matchingSchema?.Description ?? string.Empty;
+
+                yield return (response.Key, new RestApiOperationResponse(description, mediaType, matchingSchema?.ToJsonDocument()));
```

Comment:
Consider creating and using a dedicated model class with three properties - description, schema, and contentType, to represent a response schema, rather than reusing the response class. 
   
This class was designed to represent a REST API operation call result, rather than representing a model class meant to pass information from the parser to the runner. It may also be extended in future when more data (e.g a correlation vector from response headers) need to be exposed to the OpenAPI skills callers and the change should only impact RestApiOperationRunner and not the parser.

Reply:
Yeah originally I did have similar. Can you look at the `prev` in the following commit and let me know if that aligns with what you are thinking? Thanks!

https://github.com/microsoft/semantic-kernel/pull/3450/commits/86e91300867a8c702b945fd4ea513f346b80ec3e#diff-e3dd20c9a18b37db4537970a29d06152ab4792660943d6779f4697a9ec6886c0

Reply:
Also, the goal here is for the Schema to be exposed to the OpenAPI skill callers. Both so the object can validate itself or callers implement their own validation. As well as planners can represent the expectations in their manuals.

However, I do see the concern. If I understand correctly, the commit previous I shared could be improved to have the RestApiOperationResponseSchema be a member of a new/different class than RestApiOperationResponse. Will take a stab at this.

Reply:
> Yeah originally I did have similar. Can you look at the `prev` in the following commit and let me know if that aligns with what you are thinking? Thanks! See usage of `RestApiOperationResponseSchema`
> 
> [86e9130#diff-e3dd20c9a18b37db4537970a29d06152ab4792660943d6779f4697a9ec6886c0](https://github.com/microsoft/semantic-kernel/commit/86e91300867a8c702b945fd4ea513f346b80ec3e#diff-e3dd20c9a18b37db4537970a29d06152ab4792660943d6779f4697a9ec6886c0)

Yes, the RestApiOperationResponseSchema class is what we need here. However, there are a few small changes required - rename the Title property to Description, the Type property to MediaType, and remove the IsValid method. Additionally, it is unclear whether we need the Json* attributes applied to its properties.